lib/ansible/plugins/connection/ssh.py

```diff
@@ -805,26 +805,38 @@ def _file_transport_command(self, in_path, out_path, sftp_action):
                 cmd = self._build_command('sftp', to_bytes(host))
                 in_data = u"{0} {1} {2}\n".format(sftp_action, shlex_quote(in_path), shlex_quote(out_path))
                 in_data = to_bytes(in_data, nonstring='passthru')
-                (returncode, stdout, stderr) = self._run(cmd, in_data, checkrc=False)
+                try:
+                    (returncode, stdout, stderr) = self._run(cmd, in_data, checkrc=False)
+                except AnsibleConnectionFailure:
+                    returncode = 255
```

Comment:
`_ssh_retry` is at fault here, as it wraps `_run`.  And `_ssh_retry` raises AnsibleConnectionFailure`.  We can see if it is feasible to have `_ssh_retry` return the "return tuple" if it has reached it's max attempts.  I don't know how much that would affect the rest of the code though.

The current approach is pretty targeted.